src/emulator/controller.ts

```diff
@@ -225,6 +225,65 @@ export function shouldStart(options: any, name: Emulators): boolean {
   return emulatorInTargets;
 }
 
+function findExportMetadata(importPath: string): ExportMetadata | undefined {
+  const pathIsDirectory = fs.lstatSync(importPath).isDirectory();
+  if (!pathIsDirectory) {
+    return;
+  }
+
+  // If there is an export metadata file, we always prefer that
+  const importFilePath = path.join(importPath, HubExport.METADATA_FILE_NAME);
+  if (fileExistsSync(importFilePath)) {
+    return JSON.parse(fs.readFileSync(importFilePath, "utf8").toString()) as ExportMetadata;
+  }
+
+  const fileList = fs.readdirSync(importPath);
+
+  // The user might have passed a Firestore export directly
+  const firestoreMetadataFile = fileList.find((f) => f.endsWith(".overall_export_metadata"));
```

Comment:
Have we considered the following situation?

1. Developer uses CLI to export to the `/foo/bar` directory.
2. On the second run, developer forgets that `/foo/bar` contains the whole thing (and the metadata file) and specifies `--import /foo/bar/firestore_export` instead. (Maybe because the `firestore_export` name is more obvious)
3. Using the logic proposed by the PR, we'll create another metadata file at `/foo/bar/firestore_export` in addition to the one already present at `/foo/bar` and import only Firestore, but not RTDB.

Shall we address this by looking at the parent directory as well? Or shall we just let it slip?

Reply:
Also, what is the rationale behind creating an explicit metadata file? I'm not against it, but I'd love to hear why. It seems to me that things will work just fine without writing a metadata file, since the CLI always perform the check every time.

Reply:
I think that scenario is fine and we should let it slip.  The result would be the same.

As for "why create an explicit file" I struggled with this decision but I believe this is more flexible for users working with multi-emulator exports (which will be common when Auth lands).

You would be able to mix prod Firestore data with emulated Auth data by building up this explicit file over time.

Reply:
Actually now that I think about it ... that's never gonna be a good story.  I'll leave off the explicit file for now, can always add it later.